### PR TITLE
Moved Account and Profile code as per #926

### PR DESCRIPTION
**Pull request by AbhijithGanesh**

### CODE REVIEW

File: project/accounts/apps.py [hunk 1]
```diff
@@ -1,5 +1,123 @@
...
+        """
+        Gets the draft threads of the selected account
+        /accounts/{username}/drafts
+        """
+        account = get_account(username=user__username)
+        draft_threads = Thread.objects.filter(author=account, is_draft=False)
+        serializer = ThreadSerializer(
+            draft_threads, many=True, context={"request": request}
+        )
+        return Response(serializer.data)
```
Comment (brylie) [inline project/accounts/apps.py]: Let's move this viewset into `project/accounts/api.py` to indicate that it is related to the REST API.

**Review by brylie — CHANGES_REQUESTED** [review]
This looks good. Just one suggestion for the ViewSet.

File: project/accounts/apps.py (hunk 1 above)
Comment (AbhijithGanesh) [reply project/accounts/apps.py]: I shall make those changes asap.

Comment (AbhijithGanesh) [reply project/accounts/apps.py]: I fixed it as mentioned. Anything else?

File: project/accounts/apps.py [hunk 2]
```diff
@@ -1,5 +1,124 @@
...
```
Comment (brylie) [inline project/accounts/apps.py:124]: We also need to delete this code and related imports
```suggestion
```

File: project/accounts/apps.py [hunk 3]
```diff
@@ -1,5 +1,124 @@
...
+from ..permissions import IsAccountOwnerOrDuringRegistrationOrReadOnly
+
+from api.models import Thread, Account
+from api.serializers import (
+    ThreadSerializer,
+    CategorySerializer,
+    CiviSerializer,
+    AccountSerializer,
+    AccountListSerializer,
+)
```
Comment (brylie) [inline project/accounts/apps.py]: Delete unneeded imports

```suggestion
```

**Review by brylie — CHANGES_REQUESTED** [review]
We just need to clean up the remaining code from `/projects/accounts/apps.py`

File: project/accounts/api.py [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+        """
+        Gets the draft threads of the selected account
+        /accounts/{username}/drafts
+        """
+        account = get_account(username=user__username)
+        draft_threads = Thread.objects.filter(author=account, is_draft=False)
+        serializer = ThreadSerializer(
+            draft_threads, many=True, context={"request": request}
+        )
+        return Response(serializer.data)
```
Comment (brylie) [inline project/accounts/api.py]: Add a blank line at end of file
```suggestion
        return Response(serializer.data)

```

File: project/accounts/api.py [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+
```
Comment (brylie) [inline project/accounts/api.py]: Remove whitespace
```suggestion


```

**Review by brylie — CHANGES_REQUESTED** [review]
Make sure to lint the code as well, using `black`

File: project/accounts/api.py [hunk 6]
```diff
@@ -0,0 +1,222 @@
...
+from ..permissions import IsAccountOwnerOrDuringRegistrationOrReadOnly
+
+from api.models import Thread, Account
+from api.serializers import (
+    ThreadSerializer,
+    CategorySerializer,
+    CiviSerializer,
+    AccountSerializer,
+    AccountListSerializer,
+)
```
Comment (brylie) [inline project/accounts/api.py:14]: Remove unused imports

```suggestion
```

File: project/accounts/api.py [hunk 7]
```diff
@@ -0,0 +1,209 @@
+from django.apps import AppConfig
+
```
Comment (brylie) [inline project/accounts/api.py:15]: Restore imports
```suggestion
from rest_framework.viewsets import ModelViewSet
from rest_framework.decorators import action
from rest_framework.response import Response
from ..permissions import IsAccountOwnerOrDuringRegistrationOrReadOnly

from api.models import Thread, Account
from api.serializers import (
    ThreadSerializer,
    CategorySerializer,
    CiviSerializer,
    AccountSerializer,
    AccountListSerializer,
)
```

File: project/accounts/apps.py [hunk 8]
```diff
@@ -1,5 +1,112 @@
 from django.apps import AppConfig
+from rest_framework.viewsets import ModelViewSet
+
```
Comment (brylie) [inline project/accounts/apps.py]: Remove lines
```suggestion
```

File: project/accounts/apps.py [hunk 9]
```diff
@@ -1,5 +1,112 @@
...
+        """
+        Gets the draft threads of the selected account
+        /accounts/{username}/drafts
+        """
+        account = get_account(username=user__username)
+        draft_threads = Thread.objects.filter(author=account, is_draft=False)
+        serializer = ThreadSerializer(
+            draft_threads, many=True, context={"request": request}
+        )
+        return Response(serializer.data)
```
Comment (brylie) [inline project/accounts/apps.py]: Remove duplicate code
```suggestion
```

File: project/accounts/api.py [hunk 10]
```diff
@@ -0,0 +1,221 @@
...
+        """
+        Gets the draft threads of the selected account
+        /accounts/{username}/drafts
+        """
+        account = get_account(username=user__username)
+        draft_threads = Thread.objects.filter(author=account, is_draft=False)
+        serializer = ThreadSerializer(
+            draft_threads, many=True, context={"request": request}
+        )
+        return Response(serializer.data)
```
Comment (brylie) [inline project/accounts/api.py]:
```suggestion
class AccountViewSet(ModelViewSet):
    """
    REST API viewset for an Account
    retrieve:
    Return the given user based a username.
    list:
    Return a list of all the existing users. Only with privileged access.
    """

    queryset = Account.objects.all()
    lookup_field = "user__username"
    serializer_class = AccountSerializer
    http_method_names = ["get", "head", "put", "patch"]
    permission_classes = (IsAccountOwnerOrDuringRegistrationOrReadOnly,)
    authentication_classes = ()

    def list(self, request):
        """ """
        if self.request.user.is_staff:
            accounts = Account.objects.all()
        else:
            accounts = Account.objects.filter(user=self.request.user)
        serializer = AccountListSerializer(accounts, many=True)
        return Response(serializer.data)

    def retrieve(self, request, user__username=None):
        """ """
        account = get_account(username=user__username)
        if self.request.user == account.user:
            serializer = AccountSerializer(account)
        else:
            serializer = AccountListSerializer(account)
        return Response(serializer.data)

    @action(detail=True)
    def civis(self, request, user__username=None):
        """
        Gets the civis of the selected account
        /accounts/{username}/civis
        """
        account = get_account(username=user__username)
        account_civis = account.civis
        serializer = CiviSerializer(account_civis, many=True)
        return Response(serializer.data)

    @action(detail=True)
    def followers(self, request, user__username=None):
        """
        Gets the followers of the selected account
        /accounts/{username}/followers
        """
        account = get_account(username=user__username)
        account_followers = account.followers.all()
        serializer = AccountListSerializer(account_followers, many=True)
        return Response(serializer.data)

    @action(detail=True)
    def following(self, request, user__username=None):
        """
        Gets the followings of the selected account
        /accounts/{username}/following
        """
        account = get_account(username=user__username)
        account_followings = account.following.all()
        serializer = AccountListSerializer(account_followings, many=True)
        return Response(serializer.data)

    @action(detail=True)
    def categories(self, request, user__username=None):
        """
        Gets the preferred categories of the selected account
        /accounts/{username}/categories
        """
        account = get_account(username=user__username)
        account_categories = account.categories
        serializer = CategorySerializer(account_categories, many=True)
        return Response(serializer.data)

    @action(detail=True)
    def threads(self, request, user__username=None):
        """
        Gets the preferred categories of the selected account
        /accounts/{username}/categories
        """
        account = get_account(username=user__username)
        draft_threads = Thread.objects.filter(author=account).exclude(is_draft=False)
        serializer = ThreadSerializer(
            draft_threads, many=True, context={"request": request}
        )
        return Response(serializer.data)

    @action(detail=True)
    def drafts(self, request, user__username=None):
        """
        Gets the draft threads of the selected account
        /accounts/{username}/drafts
        """
        account = get_account(username=user__username)
        draft_threads = Thread.objects.filter(author=account, is_draft=False)
        serializer = ThreadSerializer(
            draft_threads, many=True, context={"request": request}
        )
        return Response(serializer.data)

```

Comment (brylie) [inline project/accounts/api.py]:
```suggestion
class AccountViewSet(ModelViewSet):
    """
    REST API viewset for an Account
    retrieve:
    Return the given user based a username.
    list:
    Return a list of all the existing users. Only with privileged access.
    """

    queryset = Account.objects.all()
    lookup_field = "user__username"
    serializer_class = AccountSerializer
    http_method_names = ["get", "head", "put", "patch"]
    permission_classes = (IsAccountOwnerOrDuringRegistrationOrReadOnly,)
    authentication_classes = ()

    def list(self, request):
        """ """
        if self.request.user.is_staff:
            accounts = Account.objects.all()
        else:
            accounts = Account.objects.filter(user=self.request.user)
        serializer = AccountListSerializer(accounts, many=True)
        return Response(serializer.data)

    def retrieve(self, request, user__username=None):
        """ """
        account = get_account(username=user__username)
        if self.request.user == account.user:
            serializer = AccountSerializer(account)
        else:
            serializer = AccountListSerializer(account)
        return Response(serializer.data)

    @action(detail=True)
    def civis(self, request, user__username=None):
        """
        Gets the civis of the selected account
        /accounts/{username}/civis
        """
        account = get_account(username=user__username)
        account_civis = account.civis
        serializer = CiviSerializer(account_civis, many=True)
        return Response(serializer.data)

    @action(detail=True)
    def followers(self, request, user__username=None):
        """
        Gets the followers of the selected account
        /accounts/{username}/followers
        """
        account = get_account(username=user__username)
        account_followers = account.followers.all()
        serializer = AccountListSerializer(account_followers, many=True)
        return Response(serializer.data)

    @action(detail=True)
    def following(self, request, user__username=None):
        """
        Gets the followings of the selected account
        /accounts/{username}/following
        """
        account = get_account(username=user__username)
        account_followings = account.following.all()
        serializer = AccountListSerializer(account_followings, many=True)
        return Response(serializer.data)

    @action(detail=True)
    def categories(self, request, user__username=None):
        """
        Gets the preferred categories of the selected account
        /accounts/{username}/categories
        """
        account = get_account(username=user__username)
        account_categories = account.categories
        serializer = CategorySerializer(account_categories, many=True)
        return Response(serializer.data)

    @action(detail=True)
    def threads(self, request, user__username=None):
        """
        Gets the preferred categories of the selected account
        /accounts/{username}/categories
        """
        account = get_account(username=user__username)
        draft_threads = Thread.objects.filter(author=account).exclude(is_draft=False)
        serializer = ThreadSerializer(
            draft_threads, many=True, context={"request": request}
        )
        return Response(serializer.data)

    @action(detail=True)
    def drafts(self, request, user__username=None):
        """
        Gets the draft threads of the selected account
        /accounts/{username}/drafts
        """
        account = get_account(username=user__username)
        draft_threads = Thread.objects.filter(author=account, is_draft=False)
        serializer = ThreadSerializer(
            draft_threads, many=True, context={"request": request}
        )
        return Response(serializer.data)

```

Comment (brylie) [inline project/accounts/api.py]:
```suggestion
```

File: project/accounts/api.py [hunk 11]
```diff
@@ -0,0 +1,120 @@
...
+from api.models import Thread, Account
+from api.serializers import (
+    ThreadSerializer,
+    CategorySerializer,
+    CiviSerializer,
+    AccountSerializer,
+    AccountListSerializer,
+)
+
+class AccountViewSet(ModelViewSet):
```
Comment (brylie) [inline project/accounts/api.py:17]: Lint
```suggestion


class AccountViewSet(ModelViewSet):

```

File: project/accounts/forms.py [hunk 12]
```diff
@@ -232,3 +232,103 @@ class Meta:
...
+            # validate file size
+            if len(profile_image) > (2000 * 1024):
+                raise forms.ValidationError(
+                    "Profile image file size may not exceed 2MB."
+                )
+
+        except AttributeError:
+            pass
+
+        return profile_image
```
Comment (brylie) [inline project/accounts/forms.py:334]:
```suggestion
        return profile_image

```

File: project/accounts/api.py [hunk 13]
```diff
@@ -0,0 +1,122 @@
...
+        Gets the draft threads of the selected account
+        /accounts/{username}/drafts
+        """
+        account = get_account(username=user__username)
+        draft_threads = Thread.objects.filter(author=account, is_draft=False)
+        serializer = ThreadSerializer(
+            draft_threads, many=True, context={"request": request}
+        )
+        return Response(serializer.data)
+
```
Comment (brylie) [inline project/accounts/api.py]: Lint
```suggestion

```

File: project/accounts/forms.py [hunk 14]
```diff
@@ -232,3 +232,104 @@ class Meta:
...
+
```
Comment (brylie) [inline project/accounts/forms.py:335]:
```suggestion
        return profile_image
```

**Review by brylie — APPROVED** [review]

**Comment by brylie** [comment]
@all-contributors please add @AbhijithGanesh for code

**Comment by AbhijithGanesh** [comment]
Thanks all, glad to be a part of community

**Pull request merged**